core/server/common/src/main/java/alluxio/AlluxioRemoteLogFilter.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import org.apache.log4j.MDC;
+import org.apache.log4j.spi.Filter;
+import org.apache.log4j.spi.LoggingEvent;
+
+/**
+ * Thin filter to add MDC information to {@link LoggingEvent}. Remote log server can read
+ * {@link org.apache.log4j.spi.LoggingEvent} off {@link java.net.Socket} and retrieve the
+ * MDC information. In this case the MDC information is the name of the log4j appender.
+ * Remote log server then uses this appender to log messages. The implementation of this
+ * class is similar to that of {@link org.apache.log4j.varia.StringMatchFilter}.
+ */
+public class AlluxioRemoteLogFilter extends Filter {
```

Comment:
Is there no constructor?

Reply:
There does not have to be a constructor here, the same for the `StringMatchFilter` provided by log4j. But I

Reply:
Explicit constructors are no longer required in the codebase

Reply:
Done.